File: src/main/java/sds/offre/tache_metier/ConsulterLesFormules.java
package sds.offre.tache_metier;

import sds.offre.concept_metier.Formule;
import sds.offre.concept_metier.FormuleRepository;

import java.util.Collection;

public class ConsulterLesFormules {
    private final FormuleRepository formuleRepository;

    public ConsulterLesFormules(FormuleRepository formuleRepository) {
        this.formuleRepository = formuleRepository;
    }

    public Collection<Formule> consulte() {
        return formuleRepository.toutesLesFormules();
    }
}

File: src/main/java/sds/offre/tache_metier/ConsulterLesFormules.kt
package sds.offre.tache_metier

import sds.offre.concept_metier.Formule
import sds.offre.concept_metier.FormuleRepository

class ConsulterLesFormules(private val formuleRepository: FormuleRepository) {

    fun consulte(): Collection<Formule> {
        return formuleRepository.toutesLesFormules()
    }
}
